### PR TITLE
Fix/vps new range

### PR DESCRIPTION
**Pull request by JeremyDec**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix many tickets, see list below
| License          | BSD 3-Clause

## Description

⚠️ PR targeting master, as it is related to VPS new range release in US.

Fixes:
* MANAGER-4967: 997497c58edecc146d3cdfe6f6409e712fcbd28b
* MANAGER-4969: d11b4294de0af30681e1b7050cfc8957f4d72b26 
* MANAGER-4894: d11b4294de0af30681e1b7050cfc8957f4d72b26
* DTRSD-11669: d17fe033dbeedce438b0972c511c2c020a8587e2
* MANAGER-4935: 30129e3fb5b3aec91504d94b880750a95fddc234
* MANAGER-5032: 2adf97e62e8bba0bfeaac8fafa5f04d37c03e966
* MANAGER-4999: 1c7a50f10d2d617ee98bf1e3636a079a4097e6cc
* MANAGER-4937: 2adf97e62e8bba0bfeaac8fafa5f04d37c03e966
* MANAGER-5087: 0d4dcb1

### CODE REVIEW

File: packages/manager/modules/vps/src/header/vps-header.controller.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ export default class {
 
   checkMessages(vps) {
     this.isExpired(vps);
-    this.isInRescueMode(vps.netbootMode);
+    this.warnForRescueMode(this.isInRescueMode);
```
Comment (FredericEspiau) [inline packages/manager/modules/vps/src/header/vps-header.controller.js:86]: `warn` ? What does this do ? It shows a dialog or a message ?

Comment (JeremyDec) [reply packages/manager/modules/vps/src/header/vps-header.controller.js:86]: It calls `CucCloudMessage.warning` to warn user that VPS is in rescue mode.
(https://github.com/ovh/manager/pull/2853/files#diff-fb48e0095e7b9f1ceac12ef1908e6342R106)

Comment (FredericEspiau) [reply packages/manager/modules/vps/src/header/vps-header.controller.js:86]: `displayWarning` instead maybe ?

File: packages/manager/modules/vps/src/additional-disk/vps-additional-disk.controller.js [hunk 2]
```diff
@@ -15,33 +14,27 @@ export default class {
       disk: false,
     };
     this.additionnalDisks = [];
-    this.hasAdditionalDiskOption = null;
   }
 
   $onInit() {
+    this.hasAdditionalDiskOption = this.tabSummary.additionalDisk.optionAvailable;
```
Comment (frenautvh) [inline packages/manager/modules/vps/src/additional-disk/vps-additional-disk.controller.js:20]: not mandatory but you could use lodash get here to be safe

**Comment by Jisay** [comment]
Good job @JeremyDec

File: packages/manager/modules/vps/src/upscale/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -45,5 +45,6 @@
   "vps_upscale_elite_upgrade_success_monthly_with_payment_method": "La modification de la configuration a bien été prise en compte. Vous avez été facturé(e) de {{ price }} au prorata temporis pour le mois en cours.",
   "vps_upscale_elite_upgrade_success_upfront_with_payment_method": "La modification de la configuration a bien été prise en compte. Vous avez été facturé(e) de {{ price }} pour le temps restant de votre engagement.",
   "vps_upscale_elite_upgrade_success_without_payment_method": "Votre <a class=\"oui-link\" target=\"_blank\" href=\"{{billUrl}}\" rel=\"noopener\">bon de commande</a> a été généré. La modification de la configuration sera effective une fois le paiement effectué.",
-  "vps_upscale_elite_upgrade_error": "Une erreur est survenue lors de la modification de la configuration."
+  "vps_upscale_elite_upgrade_error": "Une erreur est survenue lors de la modification de la configuration.",
+  "vps_upscale_agreements": "J‘accepte les <a target=\"blank\" href=\"{{conditions}}\" rel=\"noopener\">conditions générales</a> et la <a target=\"blank\" href=\"{{policy}}\" rel=\"noopener\">politique de confidentialité</a>."
```
Comment (antleblanc) [inline packages/manager/modules/vps/src/upscale/translations/Messages_fr_FR.json:49]: There is a typo, missing ˋ_ˋ (ˋ_blankˋ).

**Comment by JeremyDec** [comment]
@antleblanc / @marie-j / @Jisay, just look for the two last commits, from your previous approvals.

File: packages/manager/modules/vps/src/dashboard/terminate/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -4,5 +4,6 @@
   "vps_terminate_cancel": "Annuler",
   "vps_terminate_confirm": "Confirmer",
   "vps_terminate_success": "La suppression du service a été prise en compte, vous recevrez un e-mail de confirmation pour valider la suppression.",
-  "vps_terminate_error": "Une erreur est survenue lors de la suppression du service."
+  "vps_terminate_error": "Une erreur est survenue lors de la suppression du service.",
+  "vps_terminate_unavailable": "Vous avez souscrit cet abonnement avec un engagement de {{ monthCount }} mois. Si vous avez la moindre question, n’hésitez pas à contacter notre service client par chat ou en envoyant un Ticket d'assistance. Merci de votre compréhension"
```
Comment (marie-j) [inline packages/manager/modules/vps/src/dashboard/terminate/translations/Messages_fr_FR.json:8]:
```suggestion
  "vps_terminate_unavailable": "Vous avez souscrit cet abonnement avec un engagement de {{ monthCount }} mois. Si vous avez la moindre question, n’hésitez pas à contacter notre service client par chat ou en envoyant un ticket d'assistance. Merci de votre compréhension"
```
capital letter looks unecessary here

File: packages/manager/modules/vps/src/dashboard/terminate/vps-terminate.controller.js [hunk 5]
```diff
@@ -1,5 +1,10 @@
 export default class {
-  /* @ngInject */
+  static getDegressivityMonthDetails(degressivityInformation) {
+    const [price] = degressivityInformation.prices;
```
Comment (marie-j) [inline packages/manager/modules/vps/src/dashboard/terminate/vps-terminate.controller.js:3]: Only one price ?

File: packages/manager/modules/vps/src/dashboard/terminate/vps-terminate.routing.js [hunk 6]
```diff
@@ -38,6 +40,13 @@ export default /* @ngInject */ ($stateProvider) => {
                 ),
               ),
           ),
+      degressivityInformation: /* @ngInject */ (availableUpgrades) =>
+        availableUpgrades.find(
+          ({ prices }) =>
+            ~prices[0].pricingMode.indexOf(PRICING_MODES.DEGRESSIVITY),
```
Comment (marie-j) [inline packages/manager/modules/vps/src/dashboard/terminate/vps-terminate.routing.js:46]: A `include` doesn't suit the needs ?

File: packages/manager/modules/vps/src/dashboard/terminate/vps-terminate.controller.js (hunk 5 above)
Comment (JeremyDec) [reply packages/manager/modules/vps/src/dashboard/terminate/vps-terminate.controller.js:3]: All prices of a product have the same pricing mode. So I just need to manipulate one of them, to get degressivity information, no matter which price is used.

File: packages/manager/modules/vps/src/dashboard/terminate/vps-terminate.routing.js (hunk 6 above)
Comment (JeremyDec) [reply packages/manager/modules/vps/src/dashboard/terminate/vps-terminate.routing.js:46]: Sure, habit from manipulating 2api code

**Comment by ovh-cds** [comment]
CDS Report build#12045.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#12045.1 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report installation#12115.0 ■   
* Stage 1
  * Install ■

**Comment by ovh-cds** [comment]
CDS Report installation#12115.0 ■   
* Stage 1
  * Install ■

**Pull request merged**